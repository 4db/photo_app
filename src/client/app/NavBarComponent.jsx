import React from 'react';

class NavBarComponent extends React.Component {

    static onFavorite() {
        var count = $('#favorite').text();
        $('#favorite').text(count === '' ? 1 : parseInt(count) + 1);
    }

    static deFavorite() {
        var count = $('#favorite').text();
        $('#favorite').text(count === '1' ? '' : parseInt(count) - 1);
    }

    render() {
        return (
            <div>
                <header className="fixed-nav-bar">
                    <ul className="list">
                        <li className="item">
                            Favorite counter:&nbsp;
                        </li>
                        <li className="item" id="favorite" >
                        </li>
                    </ul>
                </header>
            </div>
        );
    }
}

export default NavBarComponent;