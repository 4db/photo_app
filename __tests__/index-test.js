import React from 'react';
import renderer from 'react-test-renderer';
import NavBarComponent from '../src/client/app/NavBarComponent.jsx';

describe('App test', () => {
    it('App renders correctly', () => {
        const tree = renderer.create(
            <div className='app'>
                <NavBarComponent />
            </div>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});