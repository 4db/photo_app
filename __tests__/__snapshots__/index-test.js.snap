exports[`App test App renders correctly 1`] = `
<div
  className="app" />
`;
